### PR TITLE
Allowing to use the LLVM intrinsic readcyclecounter from Q#

### PR DESCRIPTION
**Pull request by bettinaheim**

### CODE REVIEW

File: src/QsCompiler/Core/Dependencies.fs [hunk 1]
```diff
@@ -238,6 +239,12 @@ type BuiltIn =
     static member EnableTestingViaName =
         { FullName = { Name = "EnableTestingViaName"; Namespace = BuiltIn.DiagnosticsNamespace }; Kind = Attribute }
 
+    static member ReadCycleCounter =
+        {
+            FullName = { Name = "ReadCycleCounter"; Namespace = BuiltIn.DiagnosticsNamespace }
```
Comment (swernli) [inline src/QsCompiler/Core/Dependencies.fs:244]: I think this should be in the Core namespace or the intrinsic namespace. Diagnostic feels odd to me, though perhaps there is a logic I'm missing.

File: src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestBuiltInIntrinsics1.ll [hunk 2]
```diff
@@ -50,6 +50,7 @@ entry:
   store %String* %24, %String** %27, align 8
   call void @__quantum__rt__callable_invoke(%Callable* %19, %Tuple* %25, %Tuple* null)
   call void @__quantum__rt__callable_invoke(%Callable* %14, %Tuple* null, %Tuple* null)
+  %28 = call i64 @llvm.readcyclecounter()
```
Comment (swernli) [inline src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestBuiltInIntrinsics1.ll:53]: Awesome! So glad you got this working, I was super stuck.

**Review by swernli — APPROVED** [review]
One minor thing about namespace, I think I'd prefer Core, but I defer to you to decide what you think is best for this one.

File: src/QsCompiler/Core/Dependencies.fs (hunk 1 above)
Comment (bettinaheim) [reply src/QsCompiler/Core/Dependencies.fs:244]: I suggest to put it into a new namespace Microsoft.Quantum.LlvmIntrinsics.

**Pull request merged**